File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<>();

        for(int i = 1; i <= numRows; i++){
            List<Integer> curr = new ArrayList<>(Collections.nCopies(i, 1));

            if(i > 2){
                List<Integer> prev = result.get(result.size() -1);
                for(int n = 1; n < i-1; n++){
                    curr.set(n, prev.get(n)+ prev.get(n-1));
                }
            }
            result.add(curr);
        }
        return result;
    }
}

File: 118-pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = []
        for i in range(1, numRows+1):
            curr = [0]*i
            curr[0] = 1
            curr[-1] = 1
            if i > 2:
                prev = result[-1]
                for i in range(1, i - 1):
                    curr[i] = prev[i] + prev[i-1]
            result.append(curr[:])
        return result
                
